tests/hooks/test_bot_cascade_warning.py

```
@@ -0,0 +1,176 @@
+    for line in code_lines:
+        assert "record_fail" not in line, (
+            f"Phase 5c is warn-only; record_fail must NOT be called. "
+            f"Line: {line.strip()}"
+        )
+
+
+def test_pre_push_hook_bash_syntax_valid() -> None:
+    """Hook must parse without syntax errors after Phase 5c additions."""
+    if shutil.which("bash") is None:
+        pytest.skip("bash not available on this platform")
+    result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)